account_asset_management/models/account_asset.py

```diff
@@ -519,12 +504,73 @@ def open_entries(self):
             "domain": [("id", "in", am_ids)],
         }
 
-    @api.multi
-    def compute_depreciation_board(self):
+    def _group_lines(self, table):
         def group_lines(x, y):
             y.update({"amount": x["amount"] + y["amount"]})
             return y
 
+        # group lines prior to depreciation start period
+        depreciation_start_date = self.date_start
+        lines = table[0]["lines"]
+        lines1 = []
+        lines2 = []
+        flag = lines[0]["date"] < depreciation_start_date
+        for line in lines:
+            if flag:
+                lines1.append(line)
+                if line["date"] >= depreciation_start_date:
+                    flag = False
+            else:
+                lines2.append(line)
+        if lines1:
+            lines1 = [reduce(group_lines, lines1)]
+            lines1[0]["depreciated_value"] = 0.0
+        table[0]["lines"] = lines1 + lines2
+
+    def _compute_depreciation_line(
+        self,
+        depreciated_value_posted,
+        table_i_start,
+        line_i_start,
+        table,
+        last_line,
+        posted_lines,
+    ):
+        digits = self.env["decimal.precision"].precision_get("Account")
+
+        seq = len(posted_lines)
+        depr_line = last_line
+        last_date = table[-1]["lines"][-1]["date"]
+        depreciated_value = depreciated_value_posted
+        for entry in table[table_i_start:]:
+            for line in entry["lines"][line_i_start:]:
+                seq += 1
+                name = self._get_depreciation_entry_name(seq)
+                amount = line["amount"]
+                if line["date"] == last_date:
+                    # ensure that the last entry of the table always
+                    # depreciates the remaining value
+                    amount = self.depreciation_base - depreciated_value
+                    if self.method in ["linear-limit", "degr-limit"]:
+                        amount -= self.salvage_value
+                if amount:
+                    vals = {
+                        "previous_id": depr_line.id,
+                        "amount": round(amount, digits),
+                        "asset_id": self.id,
+                        "name": name,
+                        "line_date": line["date"],
+                        "line_days": line["days"],
+                        "init_entry": entry["init"],
+                    }
+                    depreciated_value += round(amount, digits)
+                    depr_line = self.env["account.asset.line"].create(vals)
+                else:
+                    seq -= 1
+            line_i_start = 0
+
+    def compute_depreciation_board(self):
```

Comment:
Add docstring too please.

Reply:
In this case, the method name is self-explanatory: this computes the depreciation board of the asset.